### DISCUSSION TITLE
Github Actions - general ideas

### DISCUSSION THREAD

**Discussion by dimitre** [openframeworks/openFrameworks#7048] [Ideas]
https://github.com/actions/cache
maybe using it for downloading libs? I think this can shave some time off actions.

**Comment by dimitre** [comment]
Any downside of updating Ubuntu version on actions? like 20.04 or 22.04 @ofTheo ?

**Reply by ofTheo** [reply]
@dimitre - oh yeah, def worth trying. 👍
It should be fine now we've fixed a few things with the install_dependencies script. 

I think 18.04 is getting dropped soon from the CI - so we'll need to do it sooner or later

**Comment by dimitre** [comment]
I've tested using this for apt packages and it seems great
https://github.com/marketplace/actions/cache-apt-packages
they use a fast internal mirror for apt packages so it is fast to install, and cache some files between builds too.
- https://github.com/openframeworks/openFrameworks/pull/7120

**Comment by dimitre** [comment]
Other idea here: if we use something like this
```yaml
concurrency:
  group: environment-${{ github.ref }}
  cancel-in-progress: true
```
pushing a new commit to a PR cancels the previous actions.
